ui/pages/swaps/swaps.util.js

```diff
@@ -368,6 +369,18 @@ export async function fetchToken(contractAddress, chainId) {
   return token;
 }
 
+export async function fetchTokenDecimal(contractAddress, chainId) {
+  const tokenInfo = await fetchToken(contractAddress, chainId);
+  if (tokenInfo.error) {
+    const decimalsStr = await new AssetsContractController({
```

Comment:
Controllers should not be used in the frontend, instead you'll need to ask the backend to fetch the data for you

Reply:
Usually those functions are passed via props iirc